Delete a chat message functionality

homelancer opened the pull request:
On the 'See All Messages', when the user hovers on their own sent messages, an 'x' icon shows up with a tooltip that says 'Delete Message'. Clicking on the 'x' icon will remove the message from the messages table and remove it from the sender's chat view.

For now it does not automatically refresh on the bottom chat box and the recipient message list. 

Update: July 11, 2017
Updated the messages delete feature to only update the 'status' field and prevents it from being listed in the messages list of the users in the frontend. Delete message does not physically remove the messages from the database.

Both sender and recipient messages can be deleted and also allow deleting all messages on the current view of messages.

'status' field as varchar(1) must be added to ossn_messages table in the database.

homelancer commented:
Added the message action link when sending a new message

lianglee commented:
Hi, thank you for pull request we shall take a look to changes and will revert back to you. Thanks again.

homelancer commented:
Added the auto refresh of the recipient's message list when a sender removed their own messages from their message list.
Also refreshed the chat box message list of the recipient and the sender then a message has been removed from the sender and/or recipient

githubertus commented:
Good job!
I have just verified your component on newblue.ongolito.net: Everything is working fine.

What I'd wished is the possibility to delete the messages of my conversation partner, too.
And ... while already working on that ... perhaps an option to delete ALL messages.

homelancer commented:
I thought of that - deleting also the partner's/sender's messages. But I feel like 'shushing' the sender or partner on what was said. It's like "I don't like your message so I'll delete it."

Meanwhile deleting own message is like "oops I wasn't supposed to say that so I'll delete my message".

That's the reason why I did not include the delete all messages, to avoid also deleting the sender message.

But if there are maybe at lot of people who wants to also delete the partner/sender messages and even all messages then I'll the code to it.

githubertus commented:
Yeah, there are a lot of pros and cons about that.
It's just my own experience that it doesn't happen too seldom that a conversation which started friendly in the beginning becomes really rude in the end. :)
Of course you can finally block such a user, but the bad words which were already queued up to that point will stay in place if they can't be deleted.

homelancer commented:
Ok I'm convinced. I'll look into it and add the remove sender message and delete all messages as well.

lianglee commented:
@homelancer  ,  i was thinking to do following : 

1. When user delete message,  the text is deleted,  but the item remains there....  
    To this there are few ways :
        1. change text of message with some code that indicate its delete
        2. We need to add new column to messages table called deleted so when its delete we can mark message.deleted=1

However i appreciate your work and its really useful , keep it up!

githubertus commented:
"1. When user delete message, the text is deleted, but the item remains there....

I can't confirm that: On my site the db message record is removed, too.

lianglee commented:
@githubertus i am just suggestion the feature,  the way the message delete should work.

This pull completely deletes entry from table.

I am suggesting following:

1. Only delete text not entry.

homelancer commented:
@lianglee 
I could make use of the 'viewed' field like if the value is (or bitwise checked):
0=>Not viewed or recent
1=>Viewed
2=>Not viewed and deleted
3=>Viewed and deleted
Or
We could add another fields as 'status' where the value is (or bitwise checked):
0=>Message is normally viewed or recent/new
1=>deleted (or hidden from message list)
Then we could use other value in the future where if the value is:
2=>archived (or archive normally viewed or recent/new message)
3=>archived deleted (archive deleted messages)
4=>marked as SPAM or Abuse (makes it hidden from message list)
8=>blocked messages (also hide it from the list)

I would recommend the latter by using a new field called 'status' on Ossn_Messages
For now, I'll have a version check if it is Ossn 4.x or Ossn 4.4 it will make use of the 'viewed' field

Lablnet commented:
Hellow core team why. You can not combine. Two table with each  other i means one table ossn_send_message. And other ossn__recieve_message  when anyone send message store in send table also recieve table when anyone. Reply message store in recieve table also in send table  its easily way to setting up who is seen message delete messages without effecting to other user also without remain the record database just value change . Its so change for shared hosting where there. Are more record  user delete but not deleted crash the server 
For avoid these just follow these if you want 😊

homelancer commented:
@lianglee 
There is function called markViewed($from,$to) located at /components/OssnMessages/classes/OssnMessages.php that resets the 'viewed' field to 1. It keeps resetting the 'hide' value that I set on the 'viewed' fields.
So instead of using the 'viewed' fields, I'm adding the field 'status' to Ossn_Messages table.

lianglee commented:
@homelancer  yes,  status field is good.

homelancer commented:
Updated the messages delete feature to only update the 'status' field and prevents it from being listed in the messages list of the users in the frontend. Delete message does not physically removed the messages from the database.

Both sender and recipient messages can be deleted and also allow deleting all messages on the current view of messages.
 
'status' field as varchar(1) must be added to ossn_messages table in the database.

lianglee commented:
Allow to delete message that user own #1332
